boards/nxp/mcx_n9xx_evk/board.c

```diff
@@ -0,0 +1,420 @@
+	CACHE64_CTRL0->CCR = CACHE64_CTRL_CCR_INVW1_MASK | CACHE64_CTRL_CCR_INVW0_MASK;
+	CACHE64_CTRL0->CCR |= CACHE64_CTRL_CCR_GO_MASK;
+	/* Wait until the command completes */
+	while ((CACHE64_CTRL0->CCR & CACHE64_CTRL_CCR_GO_MASK) != 0U) {
+	}
+	/* Enable cache, enable write buffer */
+	CACHE64_CTRL0->CCR = (CACHE64_CTRL_CCR_ENWRBUF_MASK | CACHE64_CTRL_CCR_ENCACHE_MASK);
+
+	/* configure reg0, reg1 to cover the whole FlexSPI
+	 * reg 0 covers the space where Zephyr resides in case of XIP from FlexSPI
+	 * reg 1 covers the storage space in case of XIP from FlexSPI
+	 */
+	CACHE64_POLSEL0->REG0_TOP = 0x7FFC00;
+	CACHE64_POLSEL0->REG1_TOP = 0x0;
+	CACHE64_POLSEL0->POLSEL =
+		(CACHE64_POLSEL_POLSEL_REG0_POLICY(1) | CACHE64_POLSEL_POLSEL_REG1_POLICY(0) |
+		 CACHE64_POLSEL_POLSEL_REG2_POLICY(0));
+
+	__ISB();
+	__DSB();
+}
+#endif
+
+void board_early_init_hook(void)
+{
+	power_mode_od();
+
+	/* Enable SCG clock */
+	CLOCK_EnableClock(kCLOCK_Scg);
+
+	/* FRO OSC setup - begin, enable the FRO for safety switching */
+
+	/* Switch to FRO 12M first to ensure we can change the clock setting */
+	CLOCK_AttachClk(kFRO12M_to_MAIN_CLK);
+
+	/* Configure Flash wait-states to support 1.2V voltage level and 150000000Hz frequency */
+	FMU0->FCTRL = (FMU0->FCTRL & ~((uint32_t)FMU_FCTRL_RWSC_MASK)) | (FMU_FCTRL_RWSC(0x3U));
+
+	/* Enable FRO HF(48MHz) output */
+	CLOCK_SetupFROHFClocking(48000000U);
+
+#ifdef CONFIG_FLASH_MCUX_FLEXSPI_XIP
+	/* Call function flexspi_clock_safe_config() to move FleXSPI clock to a stable
+	 * clock source when updating the PLL if in XIP (execute code from FlexSPI memory
+	 */
+	flexspi_clock_safe_config();
+#endif
+
+	/* Set up PLL0 */
+	const pll_setup_t pll0Setup = {
+		.pllctrl = SCG_APLLCTRL_SOURCE(1U) | SCG_APLLCTRL_SELI(27U) |
+			   SCG_APLLCTRL_SELP(13U),
+		.pllndiv = SCG_APLLNDIV_NDIV(8U),
+		.pllpdiv = SCG_APLLPDIV_PDIV(1U),
+		.pllmdiv = SCG_APLLMDIV_MDIV(50U),
+		.pllRate = 150000000U
+	};
+	/* Configure PLL0 to the desired values */
+	CLOCK_SetPLL0Freq(&pll0Setup);
+	/* PLL0 Monitor is disabled */
+	CLOCK_SetPll0MonitorMode(kSCG_Pll0MonitorDisable);
+
+	/* Switch MAIN_CLK to PLL0 */
+	CLOCK_AttachClk(kPLL0_to_MAIN_CLK);
+
+	/* Set AHBCLKDIV divider to value 1 */
+	CLOCK_SetClkDiv(kCLOCK_DivAhbClk, 1U);
+
+	CLOCK_SetupExtClocking(BOARD_XTAL0_CLK_HZ);
+
+#if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(sai0)) || DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(sai1))
+	/* < Set up PLL1 */
+	const pll_setup_t pll1_Setup = {
+		.pllctrl = SCG_SPLLCTRL_SOURCE(1U) | SCG_SPLLCTRL_SELI(3U) |
+				 SCG_SPLLCTRL_SELP(1U),
+		.pllndiv = SCG_SPLLNDIV_NDIV(25U),
+		.pllpdiv = SCG_SPLLPDIV_PDIV(10U),
+		.pllmdiv = SCG_SPLLMDIV_MDIV(256U),
+		.pllRate = 24576000U};
+
+	/* Configure PLL1 to the desired values */
+	CLOCK_SetPLL1Freq(&pll1_Setup);
+	/* Set PLL1 CLK0 divider to value 1 */
+	CLOCK_SetClkDiv(kCLOCK_DivPLL1Clk0, 1U);
+#endif
+
+#if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(flexcomm1))
```

Comment:
This looks like its copied from mcxn947 evk, and fails to enable other flexcomm clocks if enabled in DT. This cost me quite a bit of time and was highly surprising behavior. 

If I can make a bold recommendation of creating a common soc init function that any flexcomm clock if the DT node is enabled to avoid surprises?

Reply:
Thanks @teburd ,
I added all the Flexcomm clocks to this file.  FYI, one reason we stopped putting clock init code at the SOC level is because clock configuration is very app specific.  Custom boards will usually need to customize the clock config.  We moving the clock init code to the board level is easier for users.  Ideally these clocks will all be configured by devicetree.  But until we have that framework ready, we can add the Flexcomm clocks here.  Thanks